rclpy/rclpy/time.py

```diff
@@ -95,6 +95,12 @@ def __add__(self, other: Duration) -> 'Time':
     def __radd__(self, other: Duration) -> 'Time':
         return self.__add__(other)
 
+    @overload
+    def __sub__(self, other: 'Time') -> Duration: ...
+
+    @overload
+    def __sub__(self, other: Duration) -> 'Time': ...
+
     def __sub__(self, other: Union['Time', Duration]) -> Union['Time', Duration]:
```

Comment:
I notice the [overload documentation](https://docs.python.org/3/library/typing.html#overload) shows the implementation function with no type hints. Do the type hints on the implementation need to be removed so that the type checker only considers the types given by the overloads, or does it not matter?

Reply:
> The non-@overload-decorated definition, meanwhile, will be used at runtime but should be ignored by a type checker

That sentence makes it seem like the types on the implementation are ignored, so it doesn't matter if we keep these types or remove them. Mind removing them?

Reply:
Reading through mypy's section on overloads (here)[https://mypy.readthedocs.io/en/stable/more_types.html#type-checking-the-implementation] it seems required to have the  broader type annotations to type check the inside of the function which I think is probably a good idea. If you still want them removed however it seems reasonable enough